olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp

```diff
@@ -231,7 +231,13 @@ PartitionsRepository::GetPartitionsExtendedResponse(
     OLP_SDK_LOG_DEBUG_F(kLogTag,
                         "GetPartitions put to cache, hrn='%s', key='%s'",
                         catalog_str.c_str(), key.c_str());
-    cache_.Put(response.GetResult(), version, expiry, is_layer_metadata);
+    const auto put_result =
+        cache_.Put(response.GetResult(), version, expiry, is_layer_metadata);
+    if (!put_result.IsSuccessful() && request.ShouldFailOnCacheError()) {
+      OLP_SDK_LOG_ERROR_F(kLogTag, "Failed to write, hrn='%s', key='%s'",
```

Comment:
Is layer ID included in the key?

Reply:
Yes, the key here is a PartitionsRequest key with appropriate layer ID